tools/ports/icu.py

```diff
@@ -20,24 +24,74 @@ def get(ports, settings, shared):
   url = 'https://github.com/unicode-org/icu/releases/download/%s/icu4c-%s-src.zip' % (TAG, VERSION)
   ports.fetch_project('icu', url, 'icu', sha512hash=HASH)
 
-  def create(final):
-    logging.info('building port: icu')
+  def get_downloaded_icu4c_path():
+    return os.path.join(ports.get_dir(), 'icu', 'icu')
+
+  def get_icu_build_path():
+    return os.path.join(ports.get_build_dir(), 'icu')
 
-    source_path = os.path.join(ports.get_dir(), 'icu', 'icu')
-    dest_path = os.path.join(ports.get_build_dir(), 'icu')
+  def get_icu_source_path():
+    return os.path.join(get_icu_build_path(), 'source')
 
+  def prepare_build():
+    dest_path = get_icu_build_path()
+    source_path = get_downloaded_icu4c_path()
+    logging.debug(f'preparing for icu build: {source_path} -> {dest_path}')
     shutil.rmtree(dest_path, ignore_errors=True)
     shutil.copytree(source_path, dest_path)
 
-    ports.build_port(os.path.join(dest_path, 'source', 'common'), final, [os.path.join(dest_path, 'source', 'common')], ['-DU_COMMON_IMPLEMENTATION=1'])
-
-    ports.install_header_dir(os.path.join(dest_path, 'source', 'common', 'unicode'))
-
-  return [shared.Cache.get_lib('libicuuc.a', create)]
+  def build_lib(lib_output, lib_src, other_includes, build_flags):
+    logging.debug('building port: icu- ' + lib_output)
+
+    additional_build_flags = [
+        # usage of 'using namespace icu' is deprecated: icu v61
+        '-DU_USING_ICU_NAMESPACE=0',
+        # make explicit inclusion of utf header: ref utf.h
+        '-DU_NO_DEFAULT_INCLUDE_UTF_HEADERS=1',
+        # mark UnicodeString constructors explicit : ref unistr.h
+        '-DUNISTR_FROM_CHAR_EXPLICIT=explicit',
+        '-DUNISTR_FROM_STRING_EXPLICIT=explicit',
+        # generate static
+        '-DU_STATIC_IMPLEMENTATION',
+        # CXXFLAGS
+        '-std=c++11'
+    ]
+    ports.build_port(lib_src, lib_output, other_includes, build_flags + additional_build_flags)
+
+  # creator for libicu_common
+  def create_libicu_common(lib_output):
+    prepare_build()
```

Comment:
I guess the other two build function rely on the fact that this one always run first?  (They assume prepare_build has already been called?)